tests/gpflow/experimental/check_shapes/test_parser.py

```diff
@@ -745,19 +747,155 @@ def __str__(self) -> str:
+  check_shapes called at: {__file__}:{_CHECK_SHAPES_LINE}
+    Argument number (0-indexed): 0
+      Line: "a: (batch..., x)"
+                ^
+      Found unexpected character.
+""",
+        ),
+        (
+            "a: batch..., x]",
+            f"""
+Unable to parse shape specification.
+  check_shapes called at: {__file__}:{_CHECK_SHAPES_LINE}
+    Argument number (0-indexed): 0
+      Line:     "a: batch..., x]"
+                    ^
+      Expected: "["
+""",
+        ),
+        (
+            "a: [batch... x]",
+            f"""
+Unable to parse shape specification.
+  check_shapes called at: {__file__}:{_CHECK_SHAPES_LINE}
+    Argument number (0-indexed): 0
+      Line:            "a: [batch... x]"
+                                     ^
+      Expected one of: ","
+                       "]"
+""",
+        ),
+        (
+            "a: [batch..., x",
+            f"""
+Unable to parse shape specification.
+  check_shapes called at: {__file__}:{_CHECK_SHAPES_LINE}
+    Argument number (0-indexed): 0
+      Line:            "a: [batch..., x"
+                                      ^
+      Expected one of: ","
+                       "..."
+                       "]"
+""",
+        ),
+        (
+            "a: [, x]",
+            f"""
+Unable to parse shape specification.
+  check_shapes called at: {__file__}:{_CHECK_SHAPES_LINE}
+    Argument number (0-indexed): 0
+      Line:            "a: [, x]"
+                            ^
+      Expected one of: variable name (re=(?:(?:[A-Z]|[a-z])|_)(?:(?:(?:[A-Z]|[a-z])|[0-9]|_))*)
+                       "."
+                       "..."
+                       integer (re=(?:[0-9])+)
+                       "None"
+                       "]"
+                       "*"
+""",
+        ),
+        (
+            "",
+            f"""
+Unable to parse shape specification.
+  check_shapes called at: {__file__}:{_CHECK_SHAPES_LINE}
+    Argument number (0-indexed): 0
+      Line:     ""
+                 ^
+      Expected: variable name (re=(?:(?:[A-Z]|[a-z])|_)(?:(?:(?:[A-Z]|[a-z])|[0-9]|_))*)
+""",
+        ),
+        (
+            """
+  a:
+  [
+    batch...
+    x
+  ]""",
+            f"""
+Unable to parse shape specification.
+  check_shapes called at: {__file__}:{_CHECK_SHAPES_LINE}
+    Argument number (0-indexed): 0
+      Line:            "    x"
+                            ^
+      Expected one of: ","
+                       "]"
+""",
+        ),
+    ],
+)
+def test_parse_argument_spec__error(spec: str, expected_message: str) -> None:
+    with pytest.raises(SpecificationParseError) as e:
+        check_shapes(spec)
+    (message,) = e.value.args
+    assert expected_message == message
+
+
+def test_parse_and_rewrite_docstring__error() -> None:
+    # I don't actually know how to provoke an error from parsing the docstring.
+    # The problem is that *any* string is allow as free-form documentation.
+    # Update this test if you find a good example."
+    assert True
```

Comment:
So if the parser encounters a badly formatted info field does it assume everything until then must have been part of the free-form documentation? (I guess it won't matter too much as a badly formatted info string would fail to compile with sphinx anyway.)

Reply:
> So if the parser encounters a badly formatted info field does it assume everything until then must have been part of the free-form documentation? (I guess it won't matter too much as a badly formatted info string would fail to compile with sphinx anyway.)

Yes.